app/controllers/api/exchange_products_controller.rb

```diff
@@ -52,23 +67,23 @@ def load_data_from_other_params
       @incoming = params[:incoming]
     end
 
-    def render_paginated_products(products)
+    def render_paginated_products(paginated_products)
       serializer = ActiveModel::ArraySerializer.new(
-        products,
+        paginated_products,
         each_serializer: Api::Admin::ForOrderCycle::SuppliedProductSerializer,
         order_cycle: @order_cycle
       )
 
       render text: {
         products: serializer,
-        pagination: pagination_data(products)
+        pagination: pagination_data(paginated_products)
       }.to_json
     end
 
-    def pagination_data(results)
+    def pagination_data(paginated_products)
       {
-        results: results.total_count,
-        pages: results.num_pages,
+        results: paginated_products.total_count,
```

Comment:
Is this the same as the count of the new endpoint?

Reply:
no, this is products count, the new endpoint is variants count.